cpp/src/arrow/util/ubsan.h

```diff
@@ -73,7 +73,7 @@ inline std::enable_if_t<std::is_trivially_copyable_v<T> &&
                         U>
 SafeCopy(T value) {
   std::remove_const_t<U> ret;
-  std::memcpy(&ret, &value, sizeof(T));
+  std::memcpy(&ret, reinterpret_cast<const void*>(&value), sizeof(T));
```

Comment:
This one surprises me. If value is on the stack, then it is already correctly aligned for T

Reply:
Hmm, that's possible. I couldn't reproduce the original error...